<launch>
  <!-- The engine ONNX model can be optained from the release on github and added to the testdata folder. -->
  <arg name="onnx_model_path_arg" default="$(find ros_msft_onnx)/testdata/engine.onnx"/>
  <arg name="model_bounds_arg" default="[0.00, -56.00, 0.00, -80.48, -112.00, -90.82, 80.48, -112.00, -90.82,80.48, 0.00, -90.82,-80.48, 0.00, -90.82,-80.48, -112.00, 90.82,80.48, -112.00, 90.82, 80.48, 0.00, 90.82,-80.48, 0.00, 90.82]"/>
  <arg name="tracker_type_arg" default="pose"/>
  <arg name="mesh_resource_arg" default="package://ros_msft_onnx/testdata/meshes/Engine_Block.stl"/>
  <arg name="ros_msft_onnx_fake_arg" default="false"/>
  <arg name="mesh_rotation_arg" default="[0, 0, 1.57]"/>
  <arg name="confidence_arg" default=".5"/>
  <arg name="mesh_scale_arg" default="[0.001, 0.001, 0.001]"/>
  <arg name="ros_msft_onnx_link_name_arg" default="ros_msft_onnx_link"/>

  <arg name="debug_arg" default="true"/>
  <arg name="use_camera" default="true"/>
  
  <node pkg="ros_msft_onnx" type="ros_msft_onnx_node" name="ros_msft_onnx" output="screen">
    <param name="onnx_model_path" value="$(arg onnx_model_path_arg)"/>
    <param name="confidence" value="$(arg confidence_arg)"/>
    <param name="model_bounds" type="yaml" value="$(arg model_bounds_arg)"/>
    <param name="tracker_type" value="$(arg tracker_type_arg)"/>
    <param name="mesh_scale" type="yaml" value="$(arg mesh_scale_arg)"/>
    <param name="ros_msft_onnx_fake" value="$(arg ros_msft_onnx_fake_arg)"/>
    <param name="link_name" value="$(arg ros_msft_onnx_link_name_arg)"/>
    <param name="image_topic" value="/cv_camera/image_raw" />
    <param name="debug" value="$(arg debug_arg)" />
    <!-- param name="calibration" value="$(find k4a_arm_support)/calibration/ost.yaml" if="$(arg use_ros_msft_onnx)"/ -->

    <!-- param name="image_topic" value="/image_publisher/image_raw" / -->
  </node>
  
  <!-- NOTE: The image properties need to be valid for the camera, or the node will auto select the closest values -->
  <node pkg="cv_camera" type="cv_camera_node" name="cv_camera" output="screen" if="$(arg use_camera)">
    <param name="rate" type="double" value="15.0"/>
    <param name="image_width" type="double" value="640"/>
    <param name="image_height" type="double" value="480"/>
  </node>

</launch>